### PR TITLE
Added OST/DLC mappers

### PR DESCRIPTION
**Pull request by Alternity156**
Adding hooks to add OST/DLC mappers and some missing custom songs mappers.

### CODE REVIEW

File: AudicaMod/src/Hooks.cs [hunk 1]
```diff
@@ -341,5 +341,130 @@ private static void Prefix(SongListControls __instance)
             }
         }
 
+        [HarmonyPatch(typeof(SongList), "GetSong", new Type[] { typeof(string) })]
+        private static class AddToScrollerPatch
+        {
+            private static void Postfix(SongList __instance, string songID, SongList.SongData __result)
+            {
+                if (__result.songID == "destiny" ||
```
Comment (octoberU) [inline AudicaMod/src/Hooks.cs]: Use a switch for this.

File: AudicaMod/src/Hooks.cs [hunk 2]
```diff
@@ -341,5 +341,130 @@ private static void Prefix(SongListControls __instance)
             }
         }
 
+        [HarmonyPatch(typeof(SongList), "GetSong", new Type[] { typeof(string) })]
+        private static class AddToScrollerPatch
+        {
+            private static void Postfix(SongList __instance, string songID, SongList.SongData __result)
```
Comment (octoberU) [inline AudicaMod/src/Hooks.cs]: Is there no way to set these once after a song list has been loaded? `GetSong` gets called pretty frequently by the game. Having to do all these checks each time is really inefficient.

**Review by octoberU — CHANGES_REQUESTED** [review]

File: AudicaMod/src/Hooks.cs (hunk 2 above)
Comment (Alternity156) [reply AudicaMod/src/Hooks.cs]: Found a way to do it on game launch, but can't figure out how to detect when the song list finishes to refresh from the mod.

Comment (Alternity156) [reply AudicaMod/src/Hooks.cs]: This is fixed.

File: AudicaMod/src/Hooks.cs (hunk 1 above)
Comment (Alternity156) [reply AudicaMod/src/Hooks.cs]: This is changed.

**Review by octoberU — APPROVED** [review]
Looks good!

**Pull request merged**